src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/Analyzers/TargetCompatibilityReferenceAnalyzer.cs

```diff
@@ -84,17 +84,30 @@ public async Task<PackageAnalysisState> AnalyzeAsync(IProject project, PackageAn
         private async Task<NuGetReference?> GetUpdatedPackageVersionAsync(NuGetReference packageReference, IEnumerable<TargetFrameworkMoniker> targetFramework, CancellationToken token)
         {
             var latestMinorVersions = await _packageLoader.GetNewerVersionsAsync(packageReference, true, token).ConfigureAwait(false);
+            NuGetReference? prereleaseCandidate = null;
 
             foreach (var newerPackage in latestMinorVersions)
             {
                 if (await _packageLoader.DoesPackageSupportTargetFrameworksAsync(newerPackage, targetFramework, token).ConfigureAwait(false))
                 {
                     _logger.LogDebug("Package {NuGetPackage} will work on {TargetFramework}", newerPackage, targetFramework);
-                    return newerPackage;
+
+                    // Only return a pre-release version if it's the only newer major version that supports the necessary TFM
+                    if (newerPackage.IsPrerelease)
```

Comment:
I'm assuming this is not directly related to the credential provider stuff. I'm fine with it in, just wanted to make sure I'm not missing  aconnection

Reply:
yeah looks like not related to the cred provider change but the change itself to consider prerelease looks fine.

Reply:
Can we avoid this or make it a feature toggle? Some companies choose not to use pre-release software over concerns about support or viability since pre-release packages can be subject to change.

Reply:
Yea, this is unrelated. I just happened to notice while testing out my change that sometimes the latest minor/build version of a package is pre-release and we probably don't want to use those if there are other similar versions that also work on Core that aren't pre-release.

@KSchlobohm, I think this change is ok as it is because it should result in *less* pre-release packages being used. After this change, the only time pre-release packages will be used will be if there are no non-pre-release packages that support .NET Core. I think that's ok, but I see your point about enterprises not wanting to use pre-release. To mitigate that concern, I'll add a warning if we have to use a pre-release package version to get NetCore support.